src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
   // write your code here
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
+      victim.health -= 50;
+
+      if (victim.health === 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The filter operation should be performed regardless of whether the victim's health is exactly 0 or not. It should be done after every bite to ensure that any animal with health less than or equal to 0 is removed from the Animal.alive array.